### ISSUE TITLE
Generated directory paths can still have colons

### ISSUE THREAD

**Issue by nickzuber** [box/ClusterRunner#368]
I noticed in `_generate_path_from_repo_url` we remove the colons from the `netloc`, however we don't try to remove colons anywhere else like from `path`, which makes something like

```
urlparse('git@git.dev.box.net:Productivity/ClusterRunnerHealthCheck.git')
# ParseResult(scheme='', netloc='', path='git@git.dev.box.net:Productivity/ClusterRunnerHealthCheck.git', params='', query='', fragment='')
```

create a repo directory like 

```
~/.clusterrunner/repos/master/git@git.dev.box.net:Productivity/ClusterRunnerHealthCheck
```` 

Since Windows doesn't like colons in directory paths, we could just do all the illegal character removing at the end instead of trying to hit each piece as we create the path.

Something like:

```python
@staticmethod
def _generate_path_from_repo_url(base_sys_path, url):
    # ...
    return Git._clean_path_url(os.path.join(repo_directory, repo_name))

@staticmethod
def _clean_path_url(url):
    illegal_chars = [':', '-'] # any characters that we don't want in a directory path
    clean_url = ''.join(c for c in url if c not in illegal_chars)
    return clean_url
```

**Comment by tjlee0909** [comment]
I'm kind of shocked that there isn't some Python library that properly strips illegal path characters per platform that it runs on. A quick google search didn't give me anything good.

For Windows though, we aren't just limited to the the colon in terms of illegal chars, according to SO (https://stackoverflow.com/questions/1033424/how-to-remove-bad-path-characters-in-python):

```
< > : " / \ | ? *
```

**Comment by nickzuber** [comment]
@tjlee0909 Yeah we should also be careful about removing slashes `/ \` from the directory path since it'll technically have those slashes in it already as part of the path. It might be better to "clean" the path before we join the pieces then so we can remove those slashes if there happen to be any for whatever reason.
